public/utils/helpers.ts

```diff
@@ -116,35 +146,58 @@ export function dataStreamBlockedPredicate(item: Pick<DataStream, "indices">, bl
   return !!item.indices.find((dataStreamIndex) => blockedItemsSet.has(dataStreamIndex.index_name));
 }
 
-export async function filterBlockedItems<T>(
-  broswerServices: BrowserServices,
-  inputItems: T[],
-  blocksTypes: IndexOpBlocksType | IndexOpBlocksType[],
-  blocksPredicate: (item: T, blockedItemsSet: Set<string>) => boolean
-): Promise<FilteredBlockedItems<T>> {
-  const blocksTypesSet = new Set(Array.isArray(blocksTypes) ? blocksTypes : [blocksTypes]);
-  const blockedIndices = await getBlockedIndices(broswerServices);
-  // we only care about the indices with blocks type in blocksTypesSet
-  // use set to accelarate execution
-  const filteredBlockedIndicesSet = new Set(
-    Object.entries(blockedIndices)
-      .filter(
-        // blockedIndex is like this: ["index_name": ["4","5"]]
-        (blockedIndex) => blockedIndex[1].find((s) => blocksTypesSet.has(s as IndexOpBlocksType))
-        // we only take index name, do not need blocksType
-      )
-      .map((blockedIndex) => blockedIndex[0])
-  );
-  const result: FilteredBlockedItems<T> = {
+export async function filterBlockedItems(
+  browserServices: BrowserServices,
+  originInputItems: IAlias[] | DataStream[] | CatIndex[],
+  blocksTypes: INDEX_OP_BLOCKS_TYPE | INDEX_OP_BLOCKS_TYPE[],
+  indexOpTargetType: INDEX_OP_TARGET_TYPE,
+  filterRedIndices: boolean = false
+): Promise<FilteredBlockedItems> {
+  const result: FilteredBlockedItems = {
     unBlockedItems: [],
     blockedItems: [],
   };
-  inputItems.forEach((item) => {
-    if (blocksPredicate(item, filteredBlockedIndicesSet)) {
-      result.blockedItems.push(item);
-    } else {
-      result.unBlockedItems.push(item);
+  var redIndices: string[] = [];
+  if (filterRedIndices) {
+    if (!originInputItems.length && indexOpTargetType !== INDEX_OP_TARGET_TYPE.INDEX) {
```

Comment:
how about an alias point to a red index?

Reply:
If the indexOpTargetType is ALIAS, then the originInputItems shouldn't be empty beacuse we only allow "flush/refresh all" for index. If originInputItems is not empty, we will get all red indices and all blocked indices and union them to a set. Then we will filter the alias using this set.